app/views/courses/TeacherClassView.coffee

```diff
@@ -226,6 +231,17 @@ module.exports = class TeacherClassView extends RootView
     else if @state.get 'selectedCourseInstance'
       @state.set 'selectedCourseInstance', null
 
+  getSelectedCourseInstance: ->
+    unless @state.get 'selectedCourseInstance'
+      @setSelectedCourseInstance()
+    return @state.get 'selectedCourseInstance'
+
+  setSelectedCourseLockedLevel: (level) ->
```

Comment:
Should this method have a feature lock, just in case the UI appears?

i.e. `return unless me.showCourseProgressControll()`?